business/MetadataDatabase/Services/PacsService.cs

```diff
@@ -20,11 +22,35 @@ public PacsService(IOptions<PacsConfiguration> settings)
             this.settings = settings.Value;
         }
 
-        public async Task<IEnumerable<QidoSeries>> GetSeriesAsync()
+        public IEnumerable<SeriesDto> GetSeriesList()
+        {
+            Task<IEnumerable<QidoSeries>> task = this.GetSeriesListAsync();
+            task.Wait();
+            return task.Result?.ToDto();
+        }
+
+        public SeriesDto GetMetadataSeries(SeriesDto series)
+        {
+            Task<QidoSeries> fetchSeriesMetadataTask = this.GetSeriesMetadataAsync(
+                series.StudyInstanceUID,
+                series.SeriesInstanceUID);
+            fetchSeriesMetadataTask.Wait();
+            return fetchSeriesMetadataTask.Result?.ToDto();
+        }
+
+        private async Task<IEnumerable<QidoSeries>> GetSeriesListAsync()
         {
             var streamtask = client.GetStreamAsync($"http://{this.settings.Host}:{this.settings.Port}/{this.settings.Path}/series");
-            var orthancSeriesList = await JsonSerializer.DeserializeAsync<IEnumerable<QidoSeries>>(await streamtask);
-            return orthancSeriesList;
+            var pacsSeriesList = await JsonSerializer.DeserializeAsync<IEnumerable<QidoSeries>>(await streamtask);
+            return pacsSeriesList;
+        }
+
+        private async Task<QidoSeries> GetSeriesMetadataAsync(string studiesUid, string seriesUid)
+        {
+            var streamtask = client.GetStreamAsync(
+                $"http://{this.settings.Host}:{this.settings.Port}/{this.settings.Path}/studies/{studiesUid}/series/{seriesUid}/metadata");
```

Comment:
What'll happen if we switch to https in the future ? This code'll fail.

Reply:
Yes ^^

Reply:
We should add http in setting file

Reply:
Ok